src/oca_github_bot/tasks/rebase_bot.py

```diff
@@ -0,0 +1,135 @@
+    if github.github_user_can_push(gh_repo, username):
+        return True
+    modified_addon_dirs, other_changes = git_modified_addon_dirs(
+        addons_dir, target_branch
+    )
+    if other_changes or not modified_addon_dirs:
+        return False
+    # if we are modifying addons only, then the user must be maintainer of
+    # all of them on the target branch
+    current_branch = github.git_get_current_branch(cwd=addons_dir)
+    try:
+        _git_call(["git", "checkout", target_branch], cwd=addons_dir)
+        return is_maintainer(username, modified_addon_dirs)
+    finally:
+        _git_call(["git", "checkout", current_branch], cwd=addons_dir)
+
+
+@task()
+@switchable("rebase_bot")
+def rebase_bot_start(org, repo, pr, username, dry_run=False, intro_message=None):
+    with github.login() as gh:
+        gh_pr = gh.pull_request(org, repo, pr)
+        target_branch = gh_pr.base.ref
+        try:
+            if not _user_can_rebase(gh, org, repo, username, ".", target_branch):
+                github.gh_call(
+                    gh_pr.create_comment,
+                    f"Sorry @{username} you are not allowed to rebase.\n\n"
+                    f"To do so you must either have push permissions on "
+                    f"the repository, or be a declared maintainer of all "
+                    f"modified addons.\n\n"
+                    f"If you wish to adopt an addon and become it's "
+                    f"[maintainer]"
+                    f"(https://odoo-community.org/page/maintainer-role), "
+                    f"open a pull request to add "
+                    f"your GitHub login to the `maintainers` key of its "
+                    f"manifest.",
+                )
+                return
+            current_branch = github.git_get_current_branch(cwd=".")
+            if not intro_message:
+                intro_message = _get_rebase_bot_intro_message()
+            github.gh_call(
+                gh_pr.create_comment,
+                f"{intro_message}\n"
+                f"Rebasing to [{target_branch}]"
+                f"(https://github.com/{org}/{repo}/pull/{pr}/commits/{current_branch})"
+                f".",
+            )
+            # do rebase
+            _git_call(["git", "fetch", "origin", target_branch])
+            _git_call(["git", "rebase", "--autosquash", target_branch])
+            # get remote name
+            url = gh_pr.html_url
+            response = requests.get(url)
+            soup = BeautifulSoup(response.text, "html.parser")
+            if bool(soup.find_all(name="span", attrs={"class": "unknown-repo"})):
+                github.gh_call(
+                    gh_pr.create_comment,
+                    f"Sorry @{username}, impossible to rebase "
+                    f"because unknown repository.",
+                )
+                return
+            span_list = soup.find_all(
+                name="span", attrs={"class": "css-truncate-target"}
+            )
```

Comment:
Is there really no better way to find out the source branch and repo of the PR? That html parsing sounds very fragile.

Reply:
No, because https://help.github.com/en/articles/checking-out-pull-requests-locally#error-failed-to-push-some-refs.

Reply:
The GitHub API gives you access to the source branch:

![image](https://user-images.githubusercontent.com/692075/64279900-48f72300-cf50-11e9-858c-83a5e67e9926.png)

So you should be able to find the information through `gh.repository()`.

Reply:
Oh good, I will try.